File: app/src/main/java/com/pietrantuono/podcasts/interfaces/ROMEPodcastEpisodeBuilder.java
package com.pietrantuono.podcasts.interfaces;

import com.rometools.rome.feed.synd.SyndEnclosure;

import java.util.Date;
import java.util.List;

public class ROMEPodcastEpisodeBuilder {
    private String duration;
    private String author;
    private boolean isExplicit;
    private String imageUrl;
    private List<String> keywords;
    private String subtitle;
    private String summary;
    private Date pubDate;
    private String title;
    private String description;
    private List<SyndEnclosure> syndEnclosures;

    public ROMEPodcastEpisodeBuilder setDuration(String duration) {
        this.duration = duration;
        return this;
    }

    public ROMEPodcastEpisodeBuilder setAuthor(String author) {
        this.author = author;
        return this;
    }

    public ROMEPodcastEpisodeBuilder setIsExplicit(boolean isExplicit) {
        this.isExplicit = isExplicit;
        return this;
    }

    public ROMEPodcastEpisodeBuilder setImageUrl(String imageUrl) {
        this.imageUrl = imageUrl;
        return this;
    }

    public ROMEPodcastEpisodeBuilder setKeywords(List<String> keywords) {
        this.keywords = keywords;
        return this;
    }

    public ROMEPodcastEpisodeBuilder setSubtitle(String subtitle) {
        this.subtitle = subtitle;
        return this;
    }

    public ROMEPodcastEpisodeBuilder setSummary(String summary) {
        this.summary = summary;
        return this;
    }

    public ROMEPodcastEpisodeBuilder setPubDate(Date pubDate) {
        this.pubDate = pubDate;
        return this;
    }

    public ROMEPodcastEpisodeBuilder setTitle(String title) {
        this.title = title;
        return this;
    }

    public ROMEPodcastEpisodeBuilder setDescription(String description) {
        this.description = description;
        return this;
    }

    public ROMEPodcastEpisodeBuilder setEnclosures(List<SyndEnclosure> syndEnclosures) {
        this.syndEnclosures = syndEnclosures;
        return this;
    }

    public PodcastEpisodeImpl createROMEPodcastEpisode() {
        return new PodcastEpisodeImpl(duration, author, isExplicit, imageUrl, keywords, subtitle, summary, pubDate, title, description, syndEnclosures);
    }

}

File: app/src/main/java/com/pietrantuono/podcasts/interfaces/ROMEPodcastEpisodeBuilder.kt
package com.pietrantuono.podcasts.interfaces

import com.rometools.rome.feed.synd.SyndEnclosure
import java.util.*

class ROMEPodcastEpisodeBuilder {
    private var duration: String? = null
    private var author: String? = null
    private var isExplicit: Boolean? = false
    private var imageUrl: String? = null
    private var keywords: List<String>? = null
    private var subtitle: String? = null
    private var summary: String? = null
    private var pubDate: Date? = null
    private var title: String? = null
    private var description: String? = null
    private var syndEnclosures: List<SyndEnclosure>? = null
    private var link: String? = null

    fun setDuration(duration: String?): ROMEPodcastEpisodeBuilder {
        this.duration = duration
        return this
    }

    fun setAuthor(author: String?): ROMEPodcastEpisodeBuilder {
        this.author = author
        return this
    }

    fun setIsExplicit(isExplicit: Boolean?): ROMEPodcastEpisodeBuilder {
        this.isExplicit = isExplicit
        return this
    }

    fun setImageUrl(imageUrl: String?): ROMEPodcastEpisodeBuilder {
        this.imageUrl = imageUrl
        return this
    }

    fun setKeywords(keywords: List<String>?): ROMEPodcastEpisodeBuilder {
        this.keywords = keywords
        return this
    }

    fun setSubtitle(subtitle: String?): ROMEPodcastEpisodeBuilder {
        this.subtitle = subtitle
        return this
    }

    fun setSummary(summary: String?): ROMEPodcastEpisodeBuilder {
        this.summary = summary
        return this
    }

    fun setPubDate(pubDate: Date?): ROMEPodcastEpisodeBuilder {
        this.pubDate = pubDate
        return this
    }

    fun setTitle(title: String?): ROMEPodcastEpisodeBuilder {
        this.title = title
        return this
    }

    fun setDescription(description: String?): ROMEPodcastEpisodeBuilder {
        this.description = description
        return this
    }

    fun setEnclosures(syndEnclosures: List<SyndEnclosure>?): ROMEPodcastEpisodeBuilder {
        this.syndEnclosures = syndEnclosures
        return this
    }

    fun setLink(link: String?): ROMEPodcastEpisodeBuilder {
        this.link = link
        return this
    }

    fun createROMEPodcastEpisode(): RealmEpisode {
        return RealmEpisode(duration, author, isExplicit, imageUrl, keywords, subtitle, summary, pubDate, title, description, syndEnclosures, link)
    }

}